docs/proposal/observability.md

```diff
@@ -0,0 +1,305 @@
+```
+
+In addition to the above metrics, istio provides the [customMetric API](https://istio.io/latest/docs/tasks/observability/metrics/customize-metrics/) to customise the metrics and get the corresponding information from the [envoy](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes).
+
+Therefore Kmesh first needs to support istio's built-in metrics, which will be gradually improved in subsequent updates, taking into account the metrics supported by envoy.
+
+#### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+It is now clear that in order to enhance the observability of Kmesh, we need to:
+
+- Support for metrics for accesslog.
+- Support for istio's built-in metrics.
+- Support promethues scratch.
+
+#### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+Support for more envoy-supported metrics in subsequent update.
+
+### Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Because Kmesh uses eBPF, it is designed to be implemented in the kernel state. We first need to divide the metrics to be implemented into two categories, those that can be accessed in the user state and those that can be accessed in the kernel state.  Then implement the code for the actual metric collection.
+
+After getting the metrics, we need to:
+
+- Implement an `access_log filter` to send the access log to the istio control plane.
+- Integrates with Prometheus to support Prometheus scraping.
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+- **Accesslog**
+
+In istio's sidecar mode, istiod passes the access log format to envoy via the xDS.
+
+After the data plane parses the xDS to get the access log format, envoy collects and formats the access log according to the configuration.
+
+So if Kmesh needs to supplement the observability capabilities of access log, it needs to implement such a pattern as well.
+
+As shown in the follow figure, Kmesh's Ads controller passes the access log filter to the Accesslog Controller to parse out the access log format.
+
+Then according to the configuration, monitor the workload, collect the metrics to generate access log, and store the generated access log locally or send it to istiod.
```

Comment:
may be you miss some details here, how can theser metrics recorded.  How can we correlate a metric with a connection/req, i am asking becuase accesslog is one per request for http. and one per connection for tcp